src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -579,6 +609,16 @@ public static void writeTo(ScriptType scriptType, StreamOutput out) throws IOExc
                 out.writeVInt(INLINE_VAL); //Default to inline
             }
         }
+
+        @Override
+        public String toString() {
+            //ScriptModes looks for 'dynamic' rather than 'inline' when reading settings, that's the terminology
+            //we always used with users, although we prefer inline internally
+            if (this == INLINE) {
```

Comment:
it's not only a REST problem. we use this to parse the settings too. we convert the enum name to lowercase string (and to dynamic if inline). We might also print out settings to users in certain situations, in which case we do want to print out `dynamic` rather than `inline`. we could move this out of the toString but then we would have the same thing done twice, and the problem is still the same. To me this condition is there only to keep bw comp in 1.x, but we have to remove in master and make up our minds between `inline` and `dynamic`, which we cannot really do in 1.x as we have this terminology mismatch already (since we introduced INLINE instead of DYNAMIC a while ago).